core/vdbe/mod.rs

```diff
@@ -74,477 +76,553 @@ impl Display for Func {
-    BitNot {
-        reg: usize,
-        dest: usize,
-    },
-    // Jump to the instruction at address P1, P2, or P3 depending on whether in the most recent Compare instruction the P1 vector was less than, equal to, or greater than the P2 vector, respectively.
-    Jump {
-        target_pc_lt: BranchOffset,
-        target_pc_eq: BranchOffset,
-        target_pc_gt: BranchOffset,
-    },
-    // Move the P3 values in register P1..P1+P3-1 over into registers P2..P2+P3-1. Registers P1..P1+P3-1 are left holding a NULL. It is an error for register ranges P1..P1+P3-1 and P2..P2+P3-1 to overlap. It is an error for P3 to be less than 1.
-    Move {
-        source_reg: usize,
-        dest_reg: usize,
-        count: usize,
-    },
-    // If the given register is a positive integer, decrement it by decrement_by and jump to the given PC.
-    IfPos {
-        reg: usize,
-        target_pc: BranchOffset,
-        decrement_by: usize,
-    },
-    // If the given register is not NULL, jump to the given PC.
-    NotNull {
-        reg: usize,
-        target_pc: BranchOffset,
-    },
-    // Compare two registers and jump to the given PC if they are equal.
-    Eq {
-        lhs: usize,
-        rhs: usize,
-        target_pc: BranchOffset,
-    },
-    // Compare two registers and jump to the given PC if they are not equal.
-    Ne {
-        lhs: usize,
-        rhs: usize,
-        target_pc: BranchOffset,
-    },
-    // Compare two registers and jump to the given PC if the left-hand side is less than the right-hand side.
-    Lt {
-        lhs: usize,
-        rhs: usize,
-        target_pc: BranchOffset,
-    },
-    // Compare two registers and jump to the given PC if the left-hand side is less than or equal to the right-hand side.
-    Le {
-        lhs: usize,
-        rhs: usize,
-        target_pc: BranchOffset,
-    },
-    // Compare two registers and jump to the given PC if the left-hand side is greater than the right-hand side.
-    Gt {
-        lhs: usize,
-        rhs: usize,
-        target_pc: BranchOffset,
-    },
-    // Compare two registers and jump to the given PC if the left-hand side is greater than or equal to the right-hand side.
-    Ge {
-        lhs: usize,
-        rhs: usize,
-        target_pc: BranchOffset,
-    },
-    /// Jump to target_pc if r\[reg\] != 0 or (r\[reg\] == NULL && r\[null_reg\] != 0)
-    If {
-        reg: usize,              // P1
-        target_pc: BranchOffset, // P2
-        /// P3. If r\[reg\] is null, jump iff r\[null_reg\] != 0
-        null_reg: usize,
-    },
-    /// Jump to target_pc if r\[reg\] != 0 or (r\[reg\] == NULL && r\[null_reg\] != 0)
-    IfNot {
-        reg: usize,              // P1
-        target_pc: BranchOffset, // P2
-        /// P3. If r\[reg\] is null, jump iff r\[null_reg\] != 0
-        null_reg: usize,
-    },
-    // Open a cursor for reading.
-    OpenReadAsync {
-        cursor_id: CursorID,
-        root_page: PageIdx,
-    },
+enum_with_docs! {
```

Comment:
Let's do something like:

```
#[derive(Description)]
enum Insn {
```

instead to make it look more like Rust.

Btw, we could also try to make the derive macro extract description from a Rustdoc...

Reply:
' description from a Rustdoc.' I did not get this.

Reply:
+1 for using using just comments and extracting those. This does not look very rusty.

Reply:
@penberg  I have implemented proc_macro, As we know we should create lib for that to work, where exactly i should create that?

Reply:
@penberg  By rustdoc you mean I have to extract from directly from comment?
`/// Description `